commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/Mechanism.java

```diff
@@ -0,0 +1,211 @@
+        this.finalNameRewriter = builder.finalNameRewriter;
+        this.realmMapper = builder.realmMapper;
+        this.credentialSecurityFactory = builder.credentialSecurityFactory;
+        this.mechanismRealmConfigurations = builder.mechanismRealmConfigurations;
+    }
+
+    public String getMechanismName() {
+        return mechanismName;
+    }
+
+    public String getHostName() {
+        return hostName;
+    }
+
+    public String getProtocol() {
+        return protocol;
+    }
+
+    public String getPreRealmNameRewriter() {
+        return preRealmNameRewriter;
+    }
+
+    public String getPostRealmNameRewriter() {
+        return postRealmNameRewriter;
+    }
+
+    public String getFinalNameRewriter() {
+        return finalNameRewriter;
+    }
+
+    public String getRealmMapper() {
+        return realmMapper;
+    }
+
+    public String getCredentialSecurityFactory() {
+        return credentialSecurityFactory;
+    }
+
+    public List<MechanismRealm> getMechanismRealmConfigurations() {
+        return mechanismRealmConfigurations;
+    }
+
+    public static final class Builder {
+
+        private String mechanismName;
+        private String hostName;
+        private String protocol;
+        private String preRealmNameRewriter;
+        private String postRealmNameRewriter;
+        private String finalNameRewriter;
+        private String realmMapper;
+        private String credentialSecurityFactory;
+        private List<MechanismRealm> mechanismRealmConfigurations = new ArrayList<MechanismRealm>();
+
+        public Builder mechanismName(String mechanismName) {
+            this.mechanismName = mechanismName;
+            return this;
+        }
+
+        public Builder hostName(String hostName) {
+            this.hostName = hostName;
+            return this;
+        }
+
+        public Builder protocol(String protocol) {
+            this.protocol = protocol;
+            return this;
+        }
+
+        public Builder preRealmNameRewriter(String preRealmNameRewriter) {
+            this.preRealmNameRewriter = preRealmNameRewriter;
+            return this;
+        }
+
+        public Builder postRealmNameRewriter(String postRealmNameRewriter) {
+            this.postRealmNameRewriter = postRealmNameRewriter;
+            return this;
+        }
+
+        public Builder finalNameRewriter(String finalNameRewriter) {
+            this.finalNameRewriter = finalNameRewriter;
+            return this;
+        }
+
+        public Builder realmMapper(String realmMapper) {
+            this.realmMapper = realmMapper;
+            return this;
+        }
+
+        public Builder credentialSecurityFactory(String credentialSecurityFactory) {
+            this.credentialSecurityFactory = credentialSecurityFactory;
+            return this;
+        }
+
+        public Builder addMechanismRealmConfigurations(MechanismRealm... mechanismRealms) {
+            if (mechanismRealms == null) {
```

Comment:
No, this check is related to another line - Collections.addAll. Without this check calls like addMechanismRealmConfigurations(null) causes NPE. Calling addMechanismRealmConfigurations() is ok - it just do not add anything to Collection.